pkg/endpoint/providers/kube/client_test.go

```diff
@@ -27,166 +27,219 @@ var _ = Describe("Test Kube Client Provider", func() {
+		err           error
 	)
+
 	mockCtrl = gomock.NewController(GinkgoT())
 	mockKubeController = k8s.NewMockController(mockCtrl)
 	mockConfigurator = configurator.NewMockConfigurator(mockCtrl)
 
-	fakeClientSet := fake.NewSimpleClientset()
-	stopChann := make(chan struct{})
+	stopChan := make(chan struct{})
 	providerID := "provider"
 
-	cli, err := NewProvider(fakeClientSet, mockKubeController, stopChann, providerID, mockConfigurator)
-
 	mockKubeController.EXPECT().IsMonitoredNamespace(tests.BookbuyerService.Namespace).Return(true).AnyTimes()
-	Context("when testing ListEndpointsForService", func() {
-		It("verifies new Provider at context scope succeeded", func() {
-			Expect(err).To(BeNil())
-		})
 
-		It("tests GetID", func() {
-			Expect(cli.GetID()).To(Equal(providerID))
-		})
+	BeforeEach(func() {
+		fakeClientSet = fake.NewSimpleClientset()
+		provider, err = NewProvider(fakeClientSet, mockKubeController, stopChan, providerID, mockConfigurator)
+		Expect(err).ToNot(HaveOccurred())
+	})
 
-		It("should correctly return a list of endpoints for a service", func() {
-			// Should be empty for now
-			Expect(cli.ListEndpointsForService(tests.BookbuyerService)).To(BeNil())
+	It("tests GetID", func() {
+		Expect(provider.GetID()).To(Equal(providerID))
+	})
 
-			// Create bookbuyer endpoint in Bookbuyer namespace
-			endp := &corev1.Endpoints{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "Endpoints",
-					APIVersion: "v1",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name: tests.BookbuyerService.Name,
-				},
-				Subsets: []v1.EndpointSubset{
-					{
-						Addresses: []v1.EndpointAddress{
-							{
-								IP: "8.8.8.8",
-							},
+	It("should correctly return a list of endpoints for a service", func() {
+		// Should be empty for now
+		Expect(provider.ListEndpointsForService(tests.BookbuyerService)).To(BeNil())
+
+		// Create bookbuyer endpoint in Bookbuyer namespace
+		endp := &corev1.Endpoints{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Endpoints",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      tests.BookbuyerService.Name,
+				Namespace: tests.BookbuyerService.Namespace,
+			},
+			Subsets: []v1.EndpointSubset{
+				{
+					Addresses: []v1.EndpointAddress{
+						{
+							IP: "8.8.8.8",
 						},
-						Ports: []v1.EndpointPort{
-							{
-								Name:     "port",
-								Port:     88,
-								Protocol: v1.ProtocolTCP,
-							},
+					},
+					Ports: []v1.EndpointPort{
+						{
+							Name:     "port",
+							Port:     88,
+							Protocol: v1.ProtocolTCP,
 						},
 					},
 				},
-			}
+			},
+		}
 
-			_, err = fakeClientSet.CoreV1().Endpoints(tests.BookbuyerService.Namespace).
-				Create(context.TODO(), endp, metav1.CreateOptions{})
-			Expect(err).To(BeNil())
+		_, err = fakeClientSet.CoreV1().Endpoints(tests.BookbuyerService.Namespace).Create(context.TODO(), endp, metav1.CreateOptions{})
+		Expect(err).To(BeNil())
+		<-provider.GetAnnouncementsChannel()
```

Comment:
droppable announcement. Might want to move it to pubsub, but let's deal with it separately.

Reply:
@eduser25 would you mind deprecating these across the code? I haven't been keeping up much with the channels that need to be deprecated.